src/Microsoft.AspNet.Mvc.Razor/LanguageViewLocationExpander.cs

```diff
@@ -21,36 +22,32 @@ namespace RazorWebSite
     /// </example>
     public class LanguageViewLocationExpander : IViewLocationExpander
     {
-        private const string ValueKey = "language";
-        private readonly Func<ActionContext, string> _valueFactory;
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="LanguageViewLocationExpander"/>.
-        /// </summary>
-        /// <param name="valueFactory">A factory that provides tbe language to use for expansion.</param>
-        public LanguageViewLocationExpander(Func<ActionContext, string> valueFactory)
-        {
-            _valueFactory = valueFactory;
-        }
-
         /// <inheritdoc />
         public void PopulateValues(ViewLocationExpanderContext context)
         {
-            var value = _valueFactory(context.ActionContext);
-            if (!string.IsNullOrEmpty(value))
-            {
-                context.Values[ValueKey] = value;
-            }
         }
 
         /// <inheritdoc />
         public virtual IEnumerable<string> ExpandViewLocations(ViewLocationExpanderContext context,
                                                                IEnumerable<string> viewLocations)
         {
-            string value;
-            if (context.Values.TryGetValue(ValueKey, out value))
+#if DNX451
+            var cultureInfo = Thread.CurrentThread.CurrentUICulture;
```

Comment:
Has there been much discussion yet about whether we want to use `CurrentUICulture` or `CurrentCulture` for this? This feature really blurs the lines here between the two - but IMO is closer to `CurrentUICulture`. I wanted to ask the question in case it hadn't been asked.

If there's a decision here, it's worth documenting for posterity IMO.